readme.md

```diff
@@ -8,127 +8,92 @@ binaries that have been lifted to intermediate formats. Supported input formats
-    FP_out, Gamma_FP_out := R29, Gamma_R29;
-    LR_out, Gamma_LR_out := R30, Gamma_R30;
-    SP_out, Gamma_SP_out := R31, Gamma_R31;
+    assume {:captureState "%0000033f"} true;
+    goto main_basil_return;
+  main_basil_return:
+    assume {:captureState "main_basil_return"} true;
     return;
 }
+
 $ boogie boogie_out.bpl 
 
 Boogie program verifier finished with 4 verified, 0 errors
 ```
 
 ## Installation and use
 
-### Using Containers 
-
-(This method has only been tested on linux)
-
-The docker config can be used to provide bap with the asli-plugin as well as compile the tool itself.
-
-Requirements:
-
-- podman, podman-compose
-
-#### 1. Obtain the image image
-
-##### From github 
-
-`podman pull ghcr.io/uq-pac/basil-dev:latest`
-
-##### Build the images
-
-To build the images, from the root of the respository run
-
-```
-podman-compose build
-```
-
-#### 2. Use the images with compose 
-
-Individual services can be built with `podman compose build $servicename` from the root of the repo. 
-
-The services provided are:
-
-- `basil-dev` dev environment containng scala build environment, bap and cross-compilers
-   - To compile basil into the current directory using the sbt and scala provided by the docker image: 
-      - `podman compose run basil-dev sbt assembly`
-   - To enter a shell inside the container
-      - `podman compose run basil-build`
-- `basil-build`
-   - The same as above however containg only the scala build environment and compiled basil
-   - To recompile with the currently checked-out repo run `podman-compose build basil-build`
-   - To enter a shell inside the container
-      - `podman compose run basil-build`
-- `basil` precompiled jar file and tools
-   - To run the jar inside the docker image `podman-compose run basil-dev $arguments...`
-- `compiler-explorer`
-    - instance of [godbolt.org](godbolt.org) with the tools installed. 
+This is a Scala 3 project using the [mill](mill-build.com) build system, it can be run with `mill run`. 
 
-Or enter the dev container manually, mounting the current directory (the same as podman-compose basil-dev).
+### Local Development 
 
+The tool is a Scala project, and is therefore OS-independent, but is only tested under linux. 
 
-#### 2. Use the images manually  
+Furthermore, lifting input files from a given AArch64 binary is Linux-specific, and all commands given are for Linux. 
+On Windows, WSL2 may be used to run any Linux-specific tasks, but it is less thoroughly tested.
 
-##### Compiler Explorer Container
+Installing [mill](https://mill-build.com/mill/Installation_IDE_Support.html) or [sbt](https://www.scala-sbt.org/download.html) and [JDK >= 17](https://openjdk.org/install/) is required.
 
-```sh
-podman pull ghcr.io/uq-pac/basil-compiler-explorer:latest
-```
+This can be done via [coursier](https://get-coursier.io/docs/cli-overview).
 
+A mill script is checked in at `./scripts/mill` so it does not need to be installed. 
```

Comment:
You've changed this now and you should explain to use ./mill or ./mill.bat on Windows